### ISSUE TITLE
after sliding, not able to navigate the calendar to a particular date

### ISSUE THREAD

**Issue by kunalailani** [twinssbc/Ionic-Calendar#140]
Hey,
   i m not able to navigate the calendar to a particular date when i swiped forward and backward in moth view, the view is not updated.

**Comment by twinssbc** [comment]
@kunalailani when you swipe left or right, is the title changed to the target month? Which version of  ionic and  calendar are you using? Is there any error in the console?

**Comment by kunalailani** [comment]
@twinssbc Thanks for your reply. Let me rephrase my problem.
Once I swipe left or right in month view, the navigation of calendar to particular date is no more working.
But if I do the same thing before swiping month left to right, it works as expected. I m using ionic version 2.1.0 and calendar of version 0.3.7 and yes the title is changing as expected and there is no error in the console. Thank you.

**Comment by twinssbc** [comment]
@kunalailani This control only works with Ionic 1.x. If you want to use Ionic 2.x, you need to use Ionic2-Calendar.

**Comment by kunalailani** [comment]
@twinssbc sorry i thought you are talking about ionic cli version. i m using ionic v1.3.1 and calendar version  0.3.7,
sorry by mistake i have provided u with ionic CLI version.

**Comment by twinssbc** [comment]
@kunalailani Could you provide the code snippet that how do you use the calendar? For example, how do you declare the calendar element in HTML and set the binding variable in the controller?
How do you navigate to a particular date?
So you confirmed that the title is changed, what about the dates in the table? Is it changed accordingly?

**Comment by kunalailani** [comment]
@twinssbc Here is my code and please find the gif link below, that illustrate the problem more precisely.
**HTML Code**
`<ion-list>
     <ion-item ng-click="showSelectedDate()">
	 jump to
     </ion-item>			
     <ion-item class="item item-divider item-icon-left item-icon-right text-center">
	<i class="icon ion-ios-arrow-back" ng-click="prevSlide()"></i>
	{{viewTitle}}
	<i class="icon ion-ios-arrow-right" ng-click="nextSlide()"></i>							
   </ion-item>
</ion-list>
<calendar calendar-mode="month" event-source="eventSource" ng-model="selectedDate" title-changed="onViewTitleChanged(title)" monthview-display-event-template-url="monthviewDisplayEventTemplateUrl" ng-if="eventSource">
</calendar>`

**JS Code**

`$scope.selectedDate = new Date();
	var eventArry = [];
	$scope.monthviewDisplayEventTemplateUrl = 'lib/ionic-calendar/templates/custom/monthview.html';	
	$scope.getEarnCalendarData = function() {		
		reports.getEarnReports().then(function(responseObj) {		
			for (var i = 0; i < responseObj.length; i++) {				
				var year = Number(responseObj[i].date.split("-")[0]);
				var month = Number(responseObj[i].date.split("-")[1] - 1);
				var day = Number(responseObj[i].date.split("-")[2]);
				var st = new Date(Date.UTC(year, month, day));
				var et = new Date(Date.UTC(year, month, day + 1));
				var earnObj = {};				
				earnObj.startTime = st;
				earnObj.endTime = et;
				earnObj.title = responseObj[i].earn_amount.toString();
				earnObj.allDay = true;				
				eventArry.push(earnObj);				
			}			
			$scope.eventSource = eventArry;
			$scope.$broadcast('eventSourceChanged',$scope.eventSource);
		});
	};
	$scope.getEarnCalendarData();
	$scope.onViewTitleChanged = function (title) {
            $scope.viewTitle = title;            
    };
	$scope.nextSlide = function () {
		$ionicSlideBoxDelegate.next();		 	
	};
	$scope.prevSlide = function () {
		$ionicSlideBoxDelegate.previous();		 
	};
	$scope.showSelectedDate = function() {
		var options = {
		    date: new Date($scope.selectedDate),
		    mode: 'date', 		    
		    allowOldDates: true,
		    allowFutureDates: true,	    
		    androidTheme : 3,	    
		    doneButtonLabel: 'DONE',
		    doneButtonColor: '#42C7CD',
		    cancelButtonLabel: 'CANCEL',
		    cancelButtonColor: '#42C7CD'
  		};
		$cordovaDatePicker.show(options).then(function(date) {
				console.log("^^^^^^^^^^^^^^^^^" + date )
        		$scope.selectedDate = date;        		
    	}, function(error) {
    		console.log(error)
    	});
	};`

[link of gif ](https://ibb.co/f2cpo5)

**Comment by twinssbc** [comment]
@kunalailani 

If I understand the GIF correctly, the steps are as below:
1. When opening the calendar, the current date is April 19
2. Click jump to Dec 19, it goes to Dec 19
3. Click jump to Aug 19, it goes to Aug 19
4. Swipe left, it goes to July 19
5. Swipe left, it goes to June 19
6. Click jump to April 19, no change
7. Click jump to Mar 19, no change

Unfortunately I tried the same steps, not able to reproduce it.
Is it possible make the reproduce steps a bit simple?
Also when you select April 19 and Mar 19, is the selected date still printed out in the console?

**Comment by kunalailani** [comment]
@twinssbc 
Yes, I'm able to see the selected date in the console. No, there is no other way to reproduce the same.
The steps that you've mentioned above are correct, and producing error in my device.
I'm using Ionic v1.3.1, calendar v0.3.7 and using marshmallow device.

P.S: I've produced the same issue on Android L,M.

**Comment by twinssbc** [comment]
@kunalailani Is there any way to just jump and swipe once to reproduce this error?
Also how do you swipe left? By clicking the left arrow or using a real swipe left gesture?

**Comment by kunalailani** [comment]
@twinssbc ok here are the steps which also reproduces the same.
1) open the calendar
2) swipe left or right (in either way) 
3) jump to particular date

I used to swipe in both way by gesture and by clicking on left or right arrow.

**Comment by twinssbc** [comment]
@kunalailani Is this date a valid date object?
```
$scope.selectedDate = date;
```

Today I was trying to reproduce this issue, so what I did is I added a button to set the currentDate to next month as below.
It's working.
```
$scope.calendar.currentDate = new Date($scope.calendar.currentDate.setMonth($scope.calendar.currentDate.getMonth()+1));
```

But at the beginning it was not working, because the code is like below. The currentDate.setMonth actually returns the millisecond, instead of the date object.
I wonder if your code contains similar logic.
Maybe you could try not using datepicker, see if it works.
```
$scope.calendar.currentDate = $scope.calendar.currentDate.setMonth($scope.calendar.currentDate.getMonth()+1);

```

**Comment by kunalailani** [comment]
@twinssbc Thanks for your reply.
Yes,
`$scope.selectedDate = date;`
this date is a valid date as datepicker is returning date object. And i also tried,
`$scope.calendar.currentDate = new Date($scope.calendar.currentDate.setMonth($scope.calendar.currentDate.getMonth()+1));` 
this example but the same thing happens it works before swiping left or right.
Is there anything related to direction? i tried to put console in 
`scope.$on("changeDate",function(event,direction){
        console.log("direction is " + direction);
        calendarCtrl.slideView(direction)
})`
and fired the changeDate event with parameter 1 as below
`$scope.selectedDate = new Date($scope.selectedDate.setMonth($scope.selectedDate.getMonth() + 1));
 $scope.$broadcast('changeDate', 1);`
this works even after swiping left or right, but what to do in my case?

**Comment by twinssbc** [comment]
@kunalailani 
I'm thinking is it possible because below callback is treated as external event, so it doesn't trigger the Angular watch cycle? That's why you have to broadcast the change explicitly.
```
$cordovaDatePicker.show(options)
```
What if you add the changing date logic in a normal button ng-click callback?

**Comment by kunalailani** [comment]
@twinssbc Ya, i have already tried it, i have putted below code in button ng-click callback itself
`$scope.selectedDate = new Date($scope.selectedDate.setMonth($scope.selectedDate.getMonth() + 1)); `
but the same thing happens not working after swiping left or right.
It only works if i broadcast change event  with the above code in normal button ng-click callback because here it jumps to the next month and as i have provided parameter 1 
`$scope.$broadcast('changeDate', 1);`
that moves current view to next (forward).

**Comment by twinssbc** [comment]
@kunalailani I think out another reason for your issue.
I notice that you set the date like $scope.selectedDate = xxx.
But this $scope may not the same as the one used by the calendar.
Could you try a nested object, something like $scope.calendar.selectedDate = xxx?

**Comment by kunalailani** [comment]
@twinssbc Thanks for your reply.
Yes, you are correct, it works setting the date with nested object.
Thanks a lot @twinssbc  Thank you for your time :)
